### ISSUE TITLE
Include lanes into base plugin

### ISSUE THREAD

**Issue by Static-5** [anthony0br/roblox-path-plugin#6]
Another quality of life thing, allow lanes to be added through the plugin, instead of having the create a new model everytime.

**Comment by anthony0br** [comment]
Good idea. You’d probably need to work out normals of the segment lines (i.e. cross product) and translate in the direction of average normal of all lines connecting into a node. We’d need some options like Lane spacing, alignment, etc.

**Comment by arandomollie** [comment]
No need to translate entire curve - probably easier to compute centreline curve, then translate individual segments as needed and gap fill

**Comment by anthony0br** [comment]
Of course! I’m basically reinventing what ResizeAlign does, oops. I wonder if we could investigate custom alignment implementation for better performance or readability though - effectively just use the cross product of look and up vector for direction, translate by distance to the edge then use the magnitude of the difference of the two nodes to work out the length of the outer edge.

**Comment by anthony0br** [comment]
Happy for this to be implemented with this method for now though
